src/sonic-yang-models/yang-models/sonic-static-route.yang

```diff
@@ -0,0 +1,99 @@
+module sonic-static-route {
+  yang-version 1.1;
+  namespace "http://github.com/Azure/sonic-static-route";
+  prefix sroute;
+
+  import sonic-vrf {
+    prefix vrf;
+  }
+  import ietf-inet-types {
+    prefix inet;
+  }
+  import sonic-types {
+    prefix stypes;
+  }
+
+  organization
+    "SONiC";
+  contact
+    "SONiC";
+  description
+    "STATIC ROUTE yang Module for SONiC OS";
+
+  revision 2022-03-17 {
+    description
+      "First Revision";
+  }
+
+  container sonic-static-route {
+    container STATIC_ROUTE {
+      description
+        "STATIC_ROUTE part of config_db.json";
+      list STATIC_ROUTE_LIST {
+        key "vrf_name prefix";
+        leaf vrf_name {
+          type union {
+            type string {
+              pattern 'default';
+            }
+            type string {
+              pattern "Vrf[a-zA-Z0-9_-]+";
+            }
+          }
+          description
+            "Virtual Routing Instance name as key";
+        }
+        leaf prefix {
+          type inet:ip-prefix;
+          description
+            "prefix is the destination IP address, as key";
+        }
+        leaf nexthop {
+          type string {
+	  pattern '^((((25[0-5]|2[0-4][0-9]|[0-1]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[0-1]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[0-1]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[0-1]?[0-9][0-9]?)),)*((25[0-5]|2[0-4][0-9]|[0-1]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[0-1]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[0-1]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[0-1]?[0-9][0-9]?)))|((((([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])),)*)(((([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])))))$';

```

Comment:
I'm not sure this covers all combinations of IP addresses, we can better have customer validations.

Reply:
What combinations of IP addresses? 0.0.0.0 is covered. and I tested for couple of IP addresses. Anything else you have in mind?

Reply:
Will this regexp work for ipv4 embeded ipv6 addresses e.g " ::ffff:192.168.0.1" ? I'm ok if you have covered all combinations in the test cases.